SOLR-16179 : Updated the documentation as reported in the ticket , along with the examples

atarora opened the pull request:
https://issues.apache.org/jira/browse/SOLR-16179


# Description

terms.fl can be specified in the query multiple times, which is not clear from the documentation.

# Solution

Added the respective documentation with examples in JSON and XML

# Tests

Reviewed the created doc with asciidoctor and verified the changes.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [x ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

risdenk commented on solr/solr-ref-guide/modules/query-guide/pages/terms-component.adoc:
```diff
@@ -68,8 +68,57 @@ s|Required |Default: none
 +
 Specifies the field from which to retrieve terms.
 This parameter is required if `terms=true`.
+This parameter can be specified multiple times to retrieve terms for different fields.
```
Can it also be comma separated? ie: `terms.fl=title,name`

epugh replied:
Oddly enough, it can NOT be..  "Error from server at null: undefined field: \"title,name\""     It would make MORE sense to me if that was the way it was defined.

risdenk commented on solr/solr-ref-guide/modules/query-guide/pages/terms-component.adoc:
```diff
@@ -68,8 +68,57 @@ s|Required |Default: none
+  </lst>
+  <lst name="terms">
+    <lst name="name">
+    <int name="cable">8661</int>
+    <int name="sony">21</int>
```
these 2 lines should be indented?

risdenk commented on solr/solr-ref-guide/modules/query-guide/pages/terms-component.adoc:
```diff
@@ -68,8 +68,57 @@ s|Required |Default: none
+  },
+  "terms": {
+    "name": {
+    "cable": 8661,
+    "sony": 21
```
these 2 lines should be indented?

risdenk commented on solr/solr-ref-guide/modules/query-guide/pages/terms-component.adoc:
```diff
@@ -68,8 +68,57 @@ s|Required |Default: none
+    "sony": 21
+    },
+    "title": {
+    "cable": 11387,
+    "sony": 3921
```
these 2 lines should be indented?

epugh commented:
The rest of the Terms page ONLY has xml...    And since the xml output and the json output are really jsut the same, I don't think having both formats helps the readability.   I *could* see a case for just using JSON everywhere????

atarora commented:
> The rest of the Terms page ONLY has xml... And since the xml output and the json output are really jsut the same, I don't think having both formats helps the readability. I _could_ see a case for just using JSON everywhere????

I removed the JSON, but the reason I added them both as I picked the example template from terms.ttf which has examples for both XML and JSON.
